org/mozilla/jss/ssl/javax/JSSServerSocketChannel.java

```diff
@@ -0,0 +1,127 @@
+package org.mozilla.jss.ssl.javax;
+
+import java.io.*;
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.nio.channels.*;
+import java.security.*;
+import java.util.*;
+
+import javax.net.ssl.*;
+
+public class JSSServerSocketChannel extends ServerSocketChannel {
+    private JSSServerSocket sslSocket;
+    private ServerSocketChannel parent;
+    private ServerSocket parentSocket;
+    private JSSEngine engine;
+
+    public JSSServerSocketChannel(JSSServerSocket sslSocket, ServerSocketChannel parent, JSSEngine engine) throws IOException {
+        super(null);
+
+        this.sslSocket = sslSocket;
+        this.parent = parent;
+        this.parentSocket = parent.socket();
+        this.engine = engine;
+
+        // Copy blocking mode from the parent channel.
+        configureBlocking(parent.isBlocking());
+    }
+
+    public JSSServerSocketChannel(JSSServerSocket sslSocket, ServerSocket parentSocket, JSSEngine engine) throws IOException {
+        super(null);
+
+        this.sslSocket = sslSocket;
+        this.parentSocket = parentSocket;
+        this.engine = engine;
+
+        // When there is no parent channel, the channel must be in
+        // blocking mode.
+        configureBlocking(false);
+    }
+
+    public JSSSocketChannel accept() throws IOException {
+        if (parent == null) {
+            // Have to be in blocking mode. Call up to sslSocket to handle
+            // the accept. Note that JSSSocket always has a SocketChannel even
+            // when the underlying socket doesn't so the call to getChannel()
+            // will be non-null.
+            JSSSocket acceptedSocket = sslSocket.accept();
+            return acceptedSocket.getInternalChannel();
```

Comment:
@edewata What do you think about this? It isn't technically right, but if we return `null` here, we could lose a client connection. Thoughts?

Reply:
I'm not quite sure about the implication of each option, but if we don't want to deal with `parent == null` case we probably can throw an exception (either here or maybe in the constructor). That way people using JSS will know that it's not a supported configuration. As long as we provide a clear error message we don't have to worry about losing a client connection.

Reply:
Okie dokie, I'll add the exception as a later PR. Thanks!